sup3r/bias/bias_correct_means.py

```diff
@@ -0,0 +1,949 @@
+                data[feat] = (("latitude", "longitude"), out[..., i])
+        return data
+
+    def get_lat_lon(self):
+        """Get lat lon grid"""
+        with xr.open_mfdataset(self.get_height_files("January")) as res:
+            lons, lats = np.meshgrid(
+                res["longitude"].values, res["latitude"].values
+            )
+        return np.array(lats), np.array(lons)
+
+    def get_all_data(self):
+        """Get interpolated monthly means for all out heights as a dictionary
+        to use for h5 writing.
+
+        Returns
+        -------
+        out : dict
+            Dictionary of arrays containing monthly means for each hub height.
+            Also includes latitude and longitude. Spatial dimensions are
+            flattened
+        """
+        data_dict = {}
+        lats, lons = self.get_lat_lon()
+        data_dict["latitude"] = lats.flatten()[self.mask]
+        data_dict["longitude"] = lons.flatten()[self.mask]
+        s_num = len(data_dict["longitude"])
+        for i in range(1, 13):
+            month = calendar.month_name[i]
+            out = self.get_month(month)
+            for feat in self.out_features:
+                if feat not in data_dict:
+                    data_dict[feat] = np.full((s_num, 12), np.nan)
+                data = out[feat].values.flatten()[self.mask]
+                data_dict[feat][..., i - 1] = data
+        return data_dict
+
+    @property
+    def meta(self):
+        """Get a meta data dictionary on how this data is prepared"""
+        meta = {
+            "input_files": self.input_files,
+            "class": str(self.__class__),
+            "version_record": VERSION_RECORD,
+        }
+        return meta
+
+    def write_data(self, fp_out, out):
+        """Write monthly means for all heights to h5 file"""
+        if fp_out is not None:
+            if not os.path.exists(os.path.dirname(fp_out)):
+                os.makedirs(os.path.dirname(fp_out), exist_ok=True)
+
+            if not os.path.exists(fp_out) or self.overwrite:
+                with h5py.File(fp_out, "w") as f:
+                    for dset, data in out.items():
+                        f.create_dataset(dset, data=data)
+                        logger.info(f"Added {dset} to {fp_out}.")
+
+                    for k, v in self.meta.items():
+                        f.attrs[k] = json.dumps(v)
+
+                    logger.info(
+                        f"Wrote monthly means for all out heights: {fp_out}"
+                    )
+            elif os.path.exists(fp_out):
+                logger.info(f"{fp_out} already exists and overwrite=False.")
+
+    @classmethod
+    def run(
+        cls, path_pattern, in_heights, out_heights, fp_out, overwrite=False
+    ):
+        """Read vortex tif files, convert these to monthly netcdf files for all
+        input heights, interpolate this data to requested output heights, mask
+        fill values, and write all data to h5 file.
```

Comment:
Seems kind of weird that you need the netcdf intermediate step but im assuming you thought of this and need it